sql/importer: TestImportDistributedMergeStoragePrefixTracking failed

sql/importer.TestImportDistributedMergeStoragePrefixTracking [failed](https://mesolite.cluster.engflow.com/invocations/default/d5143ff1-ff32-478b-9889-bfa772c96cbd?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on master @ [ff713391ac6a0dd5d44fe68888afab012955c8b5](https://github.com/cockroachdb/cockroach/commits/ff713391ac6a0dd5d44fe68888afab012955c8b5):


```
=== RUN   TestImportDistributedMergeStoragePrefixTracking
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestImportDistributedMergeStoragePrefixTracking2893526272
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    import_stmt_test.go:5774: error executing query="IMPORT INTO t (k, v) CSV DATA ('nodelocal://1/data.csv')" args=[]: pq: failed to wait for inspect job 1149424083727679489 for table t: INSPECT found inconsistencies
        HINT: Run 'SHOW INSPECT ERRORS FOR JOB 1149424083727679489 WITH DETAILS' for more information.
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestImportDistributedMergeStoragePrefixTracking2893526272
--- FAIL: TestImportDistributedMergeStoragePrefixTracking (3.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportDistributedMergeStoragePrefixTracking.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60190

sql/importer.TestImportDistributedMergeStoragePrefixTracking [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21097515?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21097515?buildTab=artifacts#/) on master @ [f7702eb239b542c3217e64df0975e3029f28933b](https://github.com/cockroachdb/cockroach/commits/f7702eb239b542c3217e64df0975e3029f28933b):


```
=== RUN   TestImportDistributedMergeStoragePrefixTracking
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/71b342f9c5187d1b76459a36a4c30de0/logTestImportDistributedMergeStoragePrefixTracking801645646
    test_log_scope.go:82: use -show-logs to present logs inline
    import_stmt_test.go:5774: error executing query="IMPORT INTO t (k, v) CSV DATA ('nodelocal://1/data.csv')" args=[]: pq: failed to wait for inspect job 1149481599552421889 for table t: INSPECT found inconsistencies
        HINT: Run 'SHOW INSPECT ERRORS FOR JOB 1149481599552421889 WITH DETAILS' for more information.
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/71b342f9c5187d1b76459a36a4c30de0/logTestImportDistributedMergeStoragePrefixTracking801645646
--- FAIL: TestImportDistributedMergeStoragePrefixTracking (2.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportDistributedMergeStoragePrefixTracking.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/importer/import_processor.go
+++ b/pkg/sql/importer/import_processor.go
@@ -572,6 +572,8 @@ func makeIngestHelper(
 			rowStorage:            rowStorage,
 			fileAllocator:         fileAllocator,
 			baseBulkAdderProgress: baseProgress,
+			tableID:               uint64(table.Desc.ID),
+			entryCounts:           make(map[uint64]int64),
 		}
 		indexAdder = helper
 	} else {
@@ -787,12 +789,47 @@ type mergeImportBulkAdder struct {
 	fileAllocator bulksst.FileAllocator
 
 	*baseBulkAdderProgress
+
+	// tableID is the table descriptor ID, used to construct BulkOpSummaryID
+	// keys for entry counting.
+	tableID uint64
+	// currentSummaryID is the precomputed BulkOpSummaryID for the current
+	// index, updated by SetIndexID to avoid recomputing on every Add call.
+	currentSummaryID uint64
+	// entryCounts tracks the number of KV entries added per
+	// BulkOpSummaryID(tableID, indexID).
+	entryCounts map[uint64]int64
 }
 
 var _ ingestHelper = &mergeImportBulkAdder{}
 
-// SetIndexID() implements the importHelper interface.
-func (miba *mergeImportBulkAdder) SetIndexID(_ catid.IndexID) {}
+// SetIndexID implements the importHelper interface.
+func (miba *mergeImportBulkAdder) SetIndexID(indexID catid.IndexID) {
+	miba.currentSummaryID = kvpb.BulkOpSummaryID(miba.tableID, uint64(indexID))
+}
+
+// Add implements the BulkAdder interface, delegating to Writer.Add while
+// tracking entry counts per index.
+func (miba *mergeImportBulkAdder) Add(ctx context.Context, key roachpb.Key, value []byte) error {
+	if err := miba.Writer.Add(ctx, key, value); err != nil {
+		return err
+	}
+	miba.entryCounts[miba.currentSummaryID]++
+	return nil
+}
+
+// GetSummary implements the BulkAdder interface. It returns the Writer's
+// summary augmented with entry counts tracked by this adder.
+func (miba *mergeImportBulkAdder) GetSummary() kvpb.BulkOpSummary {
+	summary := miba.Writer.GetSummary()
+	if summary.EntryCounts == nil {
+		summary.EntryCounts = make(map[uint64]int64, len(miba.entryCounts))
+	}
+	for id, count := range miba.entryCounts {
+		summary.EntryCounts[id] += count
+	}
+	return summary
+}
 
 // GetFileList() implements the importHelper interface.
 func (miba *mergeImportBulkAdder) GetFileList() *bulksst.SSTFiles {

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -109,7 +109,6 @@ func TestImportDistributedMerge(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 163425)
 	dirname, cleanup := testutils.TempDir(t)
 	t.Cleanup(cleanup)
 
@@ -5743,8 +5742,6 @@ CREATE TABLE t (
 func TestImportDistributedMergeStoragePrefixTracking(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 163380)
 
 	ctx := context.Background()
 

